From 6ef4e39187e664f1c8ff774bca767514a8f3d52c Mon Sep 17 00:00:00 2001
From: Captain <openatv@gmail.com>
Date: Sat, 18 May 2024 04:56:50 +0200
Subject: [PATCH] flatbuffers-22

Use the canonical FlatBuffers scalar names accepted by the receiver build's
flatc. Keep Kodi HEAD's compressed video, memory, achievement and disc
savestate fields in sync with the original compatibility conversion.

---
 xbmc/cores/RetroPlayer/messages/savestate.fbs | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/xbmc/cores/RetroPlayer/messages/savestate.fbs b/xbmc/cores/RetroPlayer/messages/savestate.fbs
index b2d3915009..bcede78def 100644
--- a/xbmc/cores/RetroPlayer/messages/savestate.fbs
+++ b/xbmc/cores/RetroPlayer/messages/savestate.fbs
@@ -15,17 +15,17 @@ namespace KODI.RETRO.SAVESTATE;
 
 file_identifier "SAV_";
 
-enum SaveType : uint8 {
+enum SaveType : ubyte {
   Unknown,
   Auto,
   Manual
 }
 
-enum CompressionType : uint8 {
+enum CompressionType : ubyte {
   None,
 }
 
-enum DiscSlotType : uint8 {
+enum DiscSlotType : ubyte {
   Unknown,
   Disc,
   Removed
@@ -45,11 +45,11 @@ table DiscState {
 
 table Savestate {
   // Schema version
-  version:uint8 (id: 0);
+  version:ubyte (id: 0);
 
   // Savestate properties
   type:SaveType (id: 1);
-  slot:uint8 (id: 2);
+  slot:ubyte (id: 2);
   label:string (id: 3);
   caption:string (id: 11);
   created:string (id: 4); // W3C date time [ISO 8601 : 1988 (E)] with timezone info
@@ -58,8 +58,8 @@ table Savestate {
   game_file_name:string (id: 5);
 
   // Environment properties
-  timestamp_frames:uint64 (id: 6);
-  timestamp_wall_clock_ns:uint64 (id: 7);
+  timestamp_frames:ulong (id: 6);
+  timestamp_wall_clock_ns:ulong (id: 7);
 
   // Emulator properties
   emulator_addon_id:string (id: 8);
@@ -75,23 +75,23 @@ table Savestate {
   pixel_aspect_ratio:float (id: 17); // Unused
 
   // Video frame properties
-  video_data:[uint8] (id: 18);
+  video_data:[ubyte] (id: 18);
   video_width:uint16 (id: 19);
   video_height:uint16 (id: 20);
   display_aspect_ratio:float (id: 23);
   rotation_ccw:VideoRotation (id: 21);
-  video_data_compressed:[uint8] (id: 27);
+  video_data_compressed:[ubyte] (id: 27);
   video_data_compression:CompressionType = None (id: 28);
-  video_data_uncompressed_size:uint64 = 0 (id: 29);
+  video_data_uncompressed_size:ulong = 0 (id: 29);
 
   // Memory properties
-  memory_data:[uint8] (id: 10);
-  memory_data_compressed:[uint8] (id: 24);
+  memory_data:[ubyte] (id: 10);
+  memory_data_compressed:[ubyte] (id: 24);
   memory_data_compression:CompressionType = None (id: 25);
-  memory_data_uncompressed_size:uint64 = 0 (id: 26);
+  memory_data_uncompressed_size:ulong = 0 (id: 26);
 
   // Achievement properties
-  achievement_data:[uint8] (id: 30);
+  achievement_data:[ubyte] (id: 30);
 
   // Disc properties
   disc_state:DiscState (id: 31);
